File: areco_application/lib/models/task_model.dart
class TaskModel {
  final int? id;
  final String codigo;
  final String observacao;
  final DateTime dataInicio;
  final DateTime dataFim;
  final String status;
  final String atendimento;

  TaskModel({
    this.id,
    required this.codigo,
    required this.observacao,
    required this.dataInicio,
    required this.dataFim,
    required this.status,
    required this.atendimento,
  });

  factory TaskModel.fromJson(Map<String, dynamic> j) {
    return TaskModel(
      id: j['id'],
      codigo: j['codigo'],
      observacao: j['observacao'],
      dataInicio: DateTime.parse(j['dataInicio']),
      dataFim: DateTime.parse(j['dataFim']),
      status: j['status'],
      atendimento: j['atendimento'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'codigo': codigo,
      'observacao': observacao,
      'dataInicio': dataInicio.toIso8601String(),
      'dataFim': dataFim.toIso8601String(),
      'status': status,
      'atendimento': atendimento,
    };
  }
}

File: areco_api/Areco.Api/models/TaskModel.cs
using System.ComponentModel.DataAnnotations;

namespace Areco.Api.Models
{
    public class TaskModel
    {
        public int Id { get; set; }

        [Required]
        public string Codigo { get; set; }

        public string Observacao { get; set; }

        public string Atendimento { get; set; }

        [Required]
        public DateTime DataInicio { get; set; }

        [Required]
        public DateTime DataFim { get; set; }

        [Required]
        public string Status { get; set; } 
    }
}
